### PR TITLE
TechDocs: upgrade to AWS SDK v3

### PR DESCRIPTION
**Pull request by clareliguori**
Signed-off-by: Clare Liguori <liguori@amazon.com>

Upgrading to AWS SDK v3 is necessary for migrating techdocs to the proposed AWS credential provider in https://github.com/backstage/backstage/pull/14125.

Techdocs was previously upgraded to v3 and downgraded back to v2 after some missing features were discovered in v3 (see https://github.com/backstage/backstage/issues/4415), including support for chained temporary credentials and credentials from EKS service accounts.  These are both now resolved in v3, and supported by the changes in this PR.

These changes were tested locally with the following config, pulling creds from my local shared creds file:
```yaml
techdocs:
  builder: 'local'
  generator:
    runIn: 'docker'
  publisher:
    type: 'awsS3'
    awsS3:
      bucketName: 'xxxxxxxxxxxxxx'
      bucketRootPath: '/backstage-techdocs/'
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Regarding the CI failure, we've just added a dedup check are about to wrap that up in some nicer messaging when it fails. Either way, you should be able to solve it by running `yarn dedup`

Looks good, but would like a review from @backstage/techdocs-core too

File: plugins/techdocs-node/src/stages/publish/awsS3.test.ts [hunk 1]
```diff
@@ -478,7 +465,31 @@ describe('AwsS3Publish', () => {
 
       await expect(fails).rejects.toMatchObject({
         message: expect.stringMatching(
-          'TechDocs metadata fetch failed; caused by Error: Unable to read stream; caused by Error: The file invalid/triplet/path/techdocs_metadata.json does not exist',
+          'TechDocs metadata fetch failed; caused by Error: The file invalid/triplet/path/techdocs_metadata.json does not exist',
```
Comment (agentbellnorm) [inline plugins/techdocs-node/src/stages/publish/awsS3.test.ts:468]: Maybe we can update the test to expect a string containing `The file invalid/triplet/path/techdocs_metadata.json does not exist`? Seems like that is the important part.

File: plugins/techdocs-node/src/stages/publish/awsS3.ts [hunk 2]
```diff
@@ -113,11 +131,10 @@ export class AwsS3Publish implements PublisherBase {
...
       'techdocs.publisher.awsS3.credentials',
     );
-    const credentials = AwsS3Publish.buildCredentials(credentialsConfig);
-
-    // AWS Region is an optional config. If missing, default AWS env variable AWS_REGION
-    // or AWS shared credentials file at ~/.aws/credentials will be used.
-    const region = config.getOptionalString('techdocs.publisher.awsS3.region');
+    const credentials = AwsS3Publish.buildCredentials(
+      credentialsConfig,
+      region,
```
Comment (agentbellnorm) [inline plugins/techdocs-node/src/stages/publish/awsS3.ts:136]: So the credentials now depend on the region, could this lead to surprises for users? My aws knowledge is limited, but if there's something users should be aware of after this upgrade we can add it to the changeset.

**Review by agentbellnorm — COMMENTED** [review]
Thanks! 

Code looks good to me! I had two small comments.

File: plugins/techdocs-node/src/stages/publish/awsS3.test.ts (hunk 1 above)
Comment (clareliguori) [reply plugins/techdocs-node/src/stages/publish/awsS3.test.ts]: Done

File: plugins/techdocs-node/src/stages/publish/awsS3.ts (hunk 2 above)
Comment (clareliguori) [reply plugins/techdocs-node/src/stages/publish/awsS3.ts:136]: Add a note to the changeset

File: .changeset/slow-flies-marry.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/plugin-events-backend-module-aws-sqs': patch
```
Comment (Rugvip) [inline .changeset/slow-flies-marry.md:2]: Thinking it's best to split this out into a separate changeset, seeing as it's a much smaller change and unrelated to TechDocs

**Review by Rugvip — APPROVED** [review]
Thanks, LGTM! 👍

Just a minor optional nit, and I'll chase @backstage/techdocs-core a bit for review

**Review by agentbellnorm — APPROVED** [review]
Nice!

File: .changeset/slow-flies-marry.md (hunk 3 above)
Comment (Rugvip) [reply .changeset/slow-flies-marry.md]: Applied this myself so we get it shipped :grin:

**Comment by joshbranham** [comment]
I realize this just merged, but we are hitting issues with IMDSv2 not working so I was wondering when we could expect `plugin-techdocs-node@1.4.3` to be released?
